pkg/s3select/output.go

```diff
@@ -0,0 +1,529 @@
+	currentMessage.Write(messageHeaderValue)
+
+	//Now we do a CRC check on the entire messages
+	currentMessage.Write(writeCRC(currentMessage.Bytes()))
+	return currentMessage
+
+}
+
+// writeStateMessage is the function which constructs the binary message for a state message to be sent.
+func (csvOutput *Input) writeStatMessage(payload string) *bytes.Buffer {
+	var currentMessage = &bytes.Buffer{}
+	// 11 -event type - 7 - 8 "Progress" 23
+	// 13 -content-type -7 -8 "text/xml" 25
+	// 13 -message-type -7 5 "event"     22
+	//This is predefined from AMZ protocol found here: https://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectSELECTContent.html
+	// 11 -event type - 7 - 5 "Stat" 20
+	headerLen := 67
+
+	currentMessage.Write(writePayloadSize(len(payload), headerLen))
+
+	currentMessage.Write(writeHeaderSize(headerLen))
+
+	currentMessage.Write(writeCRC(currentMessage.Bytes()))
+
+	headerNameBytes := make([]byte, 1)
+	headerNameBytes[0] = 11
+	currentMessage.Write(headerNameBytes)
+
+	headerName := make([]byte, 11)
+	copy(headerName[0:], ":event-type")
+	currentMessage.Write(headerName)
+
+	headerType := make([]byte, 1)
+	headerType[0] = 7
+	currentMessage.Write(headerType)
+
+	headerNameSize := make([]byte, 2)
+
+	binary.BigEndian.PutUint16(headerNameSize, 5)
+	currentMessage.Write(headerNameSize)
+	headerValue := make([]byte, 5)
+	copy(headerValue[0:], "Stats")
+	currentMessage.Write(headerValue)
+
+	////// Creation of the Header for Content-Type 	// 13 -content-type -7 -8 "text/xml"
+
+	contentHeaderNameBytes := make([]byte, 1)
+	contentHeaderNameBytes[0] = 13
+	currentMessage.Write(contentHeaderNameBytes)
+
+	contentHeaderName := make([]byte, 13)
+	copy(contentHeaderName[0:], ":content-type")
+	currentMessage.Write(contentHeaderName)
+
+	contentHeaderType := make([]byte, 1)
+	contentHeaderType[0] = 7
+	currentMessage.Write(contentHeaderType)
+
+	contentHeaderNameSize := make([]byte, 2)
+	binary.BigEndian.PutUint16(contentHeaderNameSize, 8)
+	currentMessage.Write(contentHeaderNameSize)
+
+	contentheaderValue := make([]byte, 8)
+	copy(contentheaderValue[0:], "text/xml")
+	currentMessage.Write(contentheaderValue)
+
+	////// Creation of the Header for message-type // 13 -message-type -7 5 "event"
+
+	messageHeaderNameBytes := make([]byte, 1)
+	messageHeaderNameBytes[0] = 13
+	currentMessage.Write(messageHeaderNameBytes)
+
+	messageHeaderName := make([]byte, 13)
+	copy(messageHeaderName[0:], ":message-type")
+	currentMessage.Write(messageHeaderName)
+
+	messageHeaderType := make([]byte, 1)
+	messageHeaderType[0] = 7
+	currentMessage.Write(messageHeaderType)
+
+	messageHeaderNameSize := make([]byte, 2)
+	binary.BigEndian.PutUint16(messageHeaderNameSize, 5)
+	currentMessage.Write(messageHeaderNameSize)
+
+	messageHeaderValue := make([]byte, 5)
+	copy(messageHeaderValue[0:], "event")
+	currentMessage.Write(messageHeaderValue)
+
+	//This part is where the payload is written, this will be only one row, since we're sending one message at a types
```

Comment:
Every comment should start with `//<space><your_words....>`, additionally follow the 80 column width rule.